e2e/playwright/info-drawer/src/tests/library-properties.spec.ts

```diff
@@ -0,0 +1,270 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { expect } from '@playwright/test';
+import { ApiClientFactory, Utils, test, SitesApi, QueriesApi, SITE_VISIBILITY, SITE_ROLES } from '@alfresco/playwright-shared';
+
+test.describe('Library properties', () => {
+  let sitesApi: SitesApi;
```

Comment:
sitesApi, username, user2, user3 and site are duplicated in describe for "non manager" - can we reuse instead of duplicating?

Reply:
moved it accordingly, user2 and user3 were not needed on the top

Reply:
please let me know if I need to fix the sitesApi too (because I'm not sure how)

Reply:
@katarzynakita I was thinking about something more or less like that structure. Thanks that all these variables like sitesApi, username, user2, user3, site etc. are not duplicated and beforeEach have less duplicated code too. You can also check if rest of code like afterEach etc. won't be possible too to write like that. 

But basically in that solution you have one main describe with whole common code and two different describes for each part of e2es like "non manager" and "manager" describes. 

```ts
test.describe('Library properties', () => {
  let sitesApi: SitesApi;
  let apiClientFactory: ApiClientFactory;

  const username = `user1-${Utils.random()}`;
  const site = {
    name: `site1-${Utils.random()}`,
    id: `site-id-${Utils.random()}`,
    visibility: SITE_VISIBILITY.MODERATED,
    description: 'my site description'
  };

  test.beforeAll(async () => {
    apiClientFactory = new ApiClientFactory();
    await apiClientFactory.setUpAcaBackend('admin');
    await apiClientFactory.createUser({ username });
    sitesApi = await SitesApi.initialize(username, username);
    await sitesApi.createSite(site.name, site.visibility, site.description, site.id);
  });

  test.describe('Manager', () => {
    const siteForUpdate = {
      name: `site2-${Utils.random()}`,
      id: `site-id-${Utils.random()}`,
      visibility: SITE_VISIBILITY.MODERATED,
      description: 'my initial description'
    };
    const siteDup = `site3-${Utils.random()}`;

    test.beforeAll(async () => {
      try {
        await sitesApi.createSite(siteForUpdate.name, siteForUpdate.visibility, siteForUpdate.description, siteForUpdate.id);
        await sitesApi.createSite(siteDup);
      } catch (error) {
        console.error(`beforeAll failed: ${error}`);
      }
    });
  });

  test.describe('Non manager', () => {
    const user2 = `user2-${Utils.random()}`;
    const user3 = `user3-${Utils.random()}`;

    test.beforeAll(async () => {
      try {
        await apiClientFactory.createUser({ username: user2 });
        await apiClientFactory.createUser({ username: user3 });
        await sitesApi.addSiteMember(site.id, user2, SITE_ROLES.SITE_COLLABORATOR.ROLE);
        await sitesApi.addSiteMember(site.id, user3, SITE_ROLES.SITE_MANAGER.ROLE);
      } catch (error) {
        console.error(`beforeAll failed: ${error}`);
      }
    });
  });
});